chapters/VK_EXT_present_timing/queries.txt

```diff
@@ -0,0 +1,441 @@
+pname:pSwapchainTimeDomainCount (if this value is zero, pname:swapchain
+does not currently support display timing).
+Otherwise, pname:pSwapchainTimeDomainCount must: point to a variable set by the user
+to the number of elements in the pname:pSwapchainTimeDomains array, and on return
+the variable is overwritten with the number of values actually written to
+pname:pSwapchainTimeDomains.
+If the value of pname:pSwapchainTimeDomainCount is less than the number of
+time domains supported, at most pname:pSwapchainTimeDomainCount values will be
+written.
+If pname:pSwapchainTimeDomainCount is smaller than the number of time domains
+supported for the given pname:swapchain, ename:VK_INCOMPLETE will be returned
+instead of ename:VK_SUCCESS to indicate that not all the available values
+were returned.
+
+include::{generated}/validity/protos/vkGetSwapchainTimeDomainsEXT.txt[]
+--
+
+[open,refpage='VkSwapchainTimeDomainPropertiesEXT',desc='An available time domain for a swapchain',type='structs']
+--
+
+The sname:VkSwapchainTimeDomainPropertiesEXT structure is defined as:
+
+include::{generated}/api/structs/VkSwapchainTimeDomainPropertiesEXT.txt[]
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to an extension-specific structure.
+  * pname:timeDomain is a elink:VkTimeDomainEXT value representing a time
+    domain that is available for the swapchain.
+
+include::{generated}/validity/structs/VkSwapchainTimeDomainPropertiesEXT.txt[]
+--
+
+[open,refpage='vkSetSwapchainTimingEXT',desc='Set timing information for a swapchain',type='protos']
+--
+
+To set timing information for a swapchain, call:
+
+include::{generated}/api/protos/vkSetSwapchainTimingEXT.txt[]
+
+  * pname:device is the device associated with pname:swapchain.
+  * pname:swapchain is the swapchain to obtain timing properties for.
+  * pname:pSwapchainTimingInfo is `NULL` or a pointer to an instance of
+    the sname:VkSwapchainTimingInfoEXT structure.  If `NULL`,
+    display timing is disabled for the swapchain, otherwise enables
+    display timing and specifies which time domain to use.
+
+include::{generated}/validity/protos/vkSetSwapchainTimingEXT.txt[]
+--
+
+[open,refpage='VkSwapchainTimingInfoEXT',desc='Specify which of the available time domains to use for a swapchain',type='structs']
+--
+
+The sname:VkSwapchainTimingInfoEXT structure is defined as:
+
+include::{generated}/api/structs/VkSwapchainTimingInfoEXT.txt[]
+
+  * pname:sType is the type of this structure.
+  * pname:pNext is `NULL` or a pointer to an extension-specific structure.
+  * pname:timeDomain is a elink:VkTimeDomainEXT value representing the time
+    domain that should be used with the swapchain.
+
+include::{generated}/validity/structs/VkSwapchainTimingInfoEXT.txt[]
+
+--
+
+[open,refpage='vkGetPastPresentationTimingEXT',desc='Obtain timing of a previously-presented image',type='protos']
+--
+
+The implementation will maintain a limited amount of history of timing
```

Comment:
I'm wondering if it would useful to the app, and possible to implement, a CPU-wait that blocks until the timing info for an app-specified `presentId` has become available to query.